### ISSUE TITLE
local install complains if git version is dirty

### ISSUE THREAD

**Issue by bernstei** [libAtoms/QUIP#383]
Try to run "make install-quippy" can complain `WARNING: Built wheel for quippy-ase is invalid: Metadata 1.2 mandates PEP 440 version, but '5057832-dirty' is not.`. It seems to work, but gives the message:
```
  DEPRECATION: quippy-ase was installed using the legacy 'setup.py install' method, because a wheel could not be built for it. A possible replacement is to fix the wheel build issue reported above. Discussion can be found at https://github.com/pypa/pip/issues/8368
```
Perhaps we should massage the version string?

**Comment by jameskermode** [comment]
Good idea. If we had the tags set properly in QUIP as well as in quippy-wheels that would help, and then version string could  be something like `v0.9.7-5057832-dirty` which I think would be valid.

**Comment by bernstei** [comment]
I think that is indeed a valid PEP 440 "local" version identifier.

**Comment by Sideboard** [comment]
Looks like this now results in an error:
```
  WARNING: Built wheel for quippy-ase is invalid: Metadata 1.2 mandates PEP 440 version, but '0.9.12-31-g9671af769-dirty' is not
Failed to build quippy-ase
ERROR: Could not build wheels for quippy-ase, which is required to install pyproject.toml-based projects
```

The version string you propose does not work, see [PEP440](https://peps.python.org/pep-0440/):
```
The canonical public version identifiers MUST comply with the following scheme:
  [N!]N(.N)*[{a|b|rc}N][.postN][.devN]
⋮
Local version identifiers MUST comply with the following scheme:
  <public version identifier>[+<local version label>]
⋮
… local version labels MUST be limited to the following set of permitted characters:
    - ASCII letters ([a-zA-Z])
    - ASCII digits ([0-9])
    - periods (.)
Local version labels MUST start and end with an ASCII letter or digit.
```

So the local version identifier for the example would be `0.9.7+5057832.dirty` (starting with `v` is allowed but must be ignored for processing). But installing only worked after I had changed the version string in `$QUIP_ROOT/build/$QUIP_ARCH/VERSION` to a public identifier (`0.9.13`).

**Comment by jameskermode** [comment]
Thanks. So we just need to change the last but one `-` to a `+`?

**Comment by Sideboard** [comment]
I'm trying some things out. The `setup.py`does additional parsing that makes it a bit more confusing.

**Comment by Sideboard** [comment]
Ok, most of the details were already applied in QUIP's `setup.py`. I only had to add that all `-` are replaced by `.`. This results in a version `0.9+git0.9.12.31.g9671af769.dirty` on my laptop. Should I push that already or do we want to change it further?

**Comment by jameskermode** [comment]
Go ahead and push. Thanks!

**Comment by jameskermode** [comment]
Closed by #577
